Adding Image 4a, 4b and updating '03-3.md' accordingly (#116)
Atcold/pytorch-Deep-Learning: docs/en/week03/03-3.md

lang-ref: ch.03-3
title: Properties of natural signals
authors: Ashwin Bhola, Nyutian Long, Linfeng Zhang, and Poornima Haridas

> date: 15 February 2020

Reviewer: ```suggestion
date:
```

---

> ## [Properties of natural signals](https://www.youtube.com/watch?v=kwPWpVverkw&t=26s)

Reviewer: ```suggestion
## Properties of Natural Signals
```

If our data exhibits stationarity, locality, and compositionality, we can exploit them with networks that use sparsity, weight sharing and stacking of layers.

> ## [Exploiting properties of natural signals to build invariance and equivariance](https://www.youtube.com/watch?v=kwPWpVverkw&t=1074s)

Reviewer: ```suggestion
## Exploiting Properties of Natural Signals to Build Standard Spatial CNN
```

> ### Locality  $\Rightarrow$ sparsity

Reviewer: ```suggestion
### Locality  $\Rightarrow$ Sparsity
```

<b>Figure 5:</b> Information Representations Moving up the Hierachy</center>

> ### [Pooling](https://www.youtube.com/watch?v=kwPWpVverkw&t=2376s)

Reviewer: ```suggestion
## Pooling and Covnet - Jupyter Notebook

### Pooling
```

<b>Figure 7:</b> Pooling results </center>

> ## CNN - Jupyter Notebook

Reviewer: ```suggestion
### CNN - Jupyter Notebook
```

<center> <img src="{{site.baseurl}}/images/week03/03-3/Figure 8 Instances from the Original MNIST Dataset.png" width="350px" /><br>
<b>Figure 8:</b> Instances from the Original MNIST Dataset </center>

Before training, we normalize our data so that the initialization of the network will match our data distribution (very important!). Also, make sure that the following five operations/steps are present in your training:

 1. Feeding data to the model
 2. Computing the loss
 3. Cleaning the cache of accumulated gradients with `zero_grad()`
 4. Computing the gradients
 5. Performing a step in the optimizer method

> 

Reviewer: ```suggestion
Before training, we normalize our data so that the initialization of the network will match our data distribution (very important!). Also, make sure that the following five operations/steps are present in your training: -
1. Feeding data to the model
2. Computing the loss
3. Cleaning the cache of accumulated gradients with `zero_grad()`
4. Computing the gradients
5. Performing a step in the optimizer method
```

3. Cleaning the cache of accumulated gradients with `zero_grad()`
4. Computing the gradients
5. Performing a step in the optimizer method

> First, we train both the networks on the normalized MNIST data. The accuracy of the FC network turned out to be $87\%$ while the accuracy of the CNN turned out to be $95\%$. Given the same number of parameters, the CNN managed to train many more filters. In the FC network, filters that try to get some dependencies between things that are further away with things that are close by, are trained. They are completely wasted. Instead, in the convolutional network, all these parameters concentrate on the relationship between neighbour pixels.

Reviewer: ```suggestion

First, we train both the networks on the normalized MNIST data. The accuracy of the FC network turned out to be $87\%$ while the accuracy of the CNN turned out to be $95\%$. Given the same number of parameters, the CNN managed to train many more filters. In the FC network, filters that try to get some dependencies between things that are further away with things that are close by, are trained. They are completely wasted. Instead, in the convolutional network, all these parameters concentrate on the relationship between neighbour pixels.
```
